src/EventSourcingTests/Examples/TeleHealth/Board.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Collections.Generic;
+
+namespace EventSourcingTests.Examples.TeleHealth;
+
+#if NET6_0_OR_GREATER
+public interface IBoardEvent
+{
+    Guid BoardId { get; }
+}
+
+public class Board
+{
+    private Board()
+    {
+    }
+
+    public Board(BoardOpened opened)
+    {
+        Name = opened.Name;
+        Activated = opened.Opened;
+        Date = opened.Date;
+    }
+
+    public void Apply(BoardFinished finished)
+    {
+        Finished = finished.Timestamp;
+    }
+
+    public void Apply(BoardClosed closed)
+    {
+        Closed = closed.Timestamp;
+        CloseReason = closed.Reason;
+    }
+
+    public Guid Id { get; set; }
+    public string Name { get; private set; }
+    public DateTimeOffset Activated { get; set; }
+    public DateTimeOffset? Finished { get; set; }
+    public DateOnly Date { get; set; }
+    public DateTimeOffset? Closed { get; set; }
+
+    public string CloseReason { get; private set; }
+}
+
+internal interface BoardStateEvent{}
```

Comment:
I'm not a huge fan of interfaces for events. I see that it can be useful, but I did that in the past and saw that folks are trying to do when they're starting to use Event Sourcing/
Too often, that leads to wrong event modelling, as they're still thinking about making all things generic instead of treating each event as a different fact.
Especially putting `State` in the name is dangerous in samples, as already too many folks are trying just to reflect state changes instead of recording the business fact.

See also:
- https://event-driven.io/en/state-obsession/
- https://event-driven.io/en/property-sourcing/
That leads too often to CRUD sourcing and/or state obsession